src/Compilers/Core/Portable/Operations/Operation.cs

```diff
@@ -22,7 +22,8 @@ internal abstract class Operation : IOperation
 
         // this will be lazily initialized. this will be initialized only once
         // but once initialized, will never change
-        private IOperation _parentDoNotAccessDirectly;
+        private IOperation _parentDoNotAccessDirectly = s_sentinel;
+        private static readonly IOperation s_sentinel = new EmptyStatement(null, null, null, default, isImplicit: true); 
```

Comment:
What's the purpose of using a sentinel value rather than `null`? The old code seemed fine.
If we're going to have a sentinel, should it be a dedicated `Operation` that throws on every API?

Reply:
> What's the purpose of using a sentinel value rather than null? The old code seemed fine.

It only seemed fine. Null is a valid value for the parent. Therefore, when the parent was null, all the checks below failed to prevent us from doing the work over and over again. And, if SemanticModel is not set, doing that work actually causes a failure. Using non-null sentinel is a common pattern in our codebase for situations like this.

> If we're going to have a sentinel, should it be a dedicated Operation that throws on every API?

We never return the sentinel, so I do not believe doing this is useful.

---
In reply to: [163065193](https://github.com/dotnet/roslyn/pull/24358#discussion_r163065193) [](ancestors = 163065193)

Reply:
Ok. Makes sense. I just saw the test that expects `null` Parent. Thanks

---
In reply to: [163068775](https://github.com/dotnet/roslyn/pull/24358#discussion_r163068775) [](ancestors = 163068775,163065193)